### PR TITLE
EthGasStationProvider

### PR DESCRIPTION
**Pull request by mikeshultz**
### Description:

Adds EthGasStationProvider subprovider to @origin/web3-provider that pulls current safeLow gas price from ethgasstation.info
 
### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

File: packages/web3-provider/src/subproviders/EthGasStationProvider.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            data,
+            time: now
+          }
+
+          return end(null, data)
+        })
+        .catch(err => {
+          console.error('Error fetching results from ethgastation.info')
+          console.error(err)
+          return next()
```
Comment (nick) [inline packages/web3-provider/src/subproviders/EthGasStationProvider.js]: Do we want to fall back if ethgasstation is down for some reason?

File: packages/web3-provider/src/subproviders/EthGasStationProvider.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  async fetchData() {
+    const res = await fetch(GAS_STATION_URL)
+    if (res.status !== 200) {
+      throw new Error(`Fetch returned code ${res.status}`)
+    }
+    const jason = await res.json()
+    if (typeof jason[GAS_PRICE_KEY] !== 'undefined') {
+      // values come from EGS as tenths of gwei
+      return String(jason[GAS_PRICE_KEY] * 1e8)
```
Comment (nick) [inline packages/web3-provider/src/subproviders/EthGasStationProvider.js]: These numbers won't ever be so big as to need BN?

File: packages/web3-provider/src/subproviders/EthGasStationProvider.js (hunk 1 above)
Comment (mikeshultz) [reply packages/web3-provider/src/subproviders/EthGasStationProvider.js]: `next()` should allow the following subprovider(s) to handle it, essentially skipping this one.  In this case, it'll be ThrottleRPCProvider, which will send it along to the JSON-RPC provider.

File: packages/web3-provider/src/subproviders/EthGasStationProvider.js (hunk 2 above)
Comment (mikeshultz) [reply packages/web3-provider/src/subproviders/EthGasStationProvider.js]: If gas price is ever at 9 Ether(max JS int) we have other issues.  Think this should throw in that case?  I really don't foresee a near future where that could be real.

Comment (nick) [reply packages/web3-provider/src/subproviders/EthGasStationProvider.js]: Ah yea no need to if it's 9 Eth

**Comment by mikeshultz** [comment]
This should probably not be enabled on networks other than mainnet...

File: packages/web3-provider/src/subproviders/EthGasStationProvider.js (hunk 1 above)
Comment (franckc) [reply packages/web3-provider/src/subproviders/EthGasStationProvider.js]: nit: I had the same question as nick. So maybe add a comment explaining next will cause default web3 gas price implementation to be used. It was not directly obvious to me neither.

**Pull request merged**